# Dow Jones Index : Correlation
### - John Antony (W1189757)

## Importing Libraries

In [219]:
import pandas as pd
from pandas import DataFrame as df
import pandas.io.data as web
import pandas_datareader.data as web
import datetime  
import matplotlib.pyplot as plt
import numpy as nm
%pylab inline

Populating the interactive namespace from numpy and matplotlib


##### Trial to get ticker infromation with a start and end date

In [220]:
# start = datetime.datetime(2013, 1, 1)
# end = datetime.datetime(2016, 1, 27)
# df = web.DataReader("GOOGL", 'yahoo', start, end)

# dates =[]
# for x in range(len(df)):
#     newdate = str(df.index[x])
#     newdate = newdate[0:10]
#     dates.append(newdate)

# df['dates'] = dates

# print df.head()
# print df.tail()

In [221]:
# all_data = {}
# for ticker in ['AAPL', 'IBM', 'MSFT', 'GOOG']:
#     try:
#         all_data[ticker] = web.get_data_yahoo(ticker, '1/1/2003', '1/1/2013')
#         price = DataFrame({tic: data['Adj Close']
#                     for tic, data in all_data.iteritems()})
#         volume = DataFrame({tic: data['Volume']
#                     for tic, data in all_data.iteritems()})
#     except:
#         print "Cant find ", ticker

### Checking Ticket and Saving for - DJIA

In [222]:
tickdata={}

In [223]:
for ticket in ['DJIA']:
    tickdata[ticket]=web.get_data_yahoo(ticket)
    
djia=df({tic:data['Adj Close']
                for tic, data in tickdata.iteritems()})

In [224]:
djia.head()

DJIA
Date                    
2010-01-04  10583.959961
2010-01-05  10572.019531
2010-01-06  10573.679688
2010-01-07  10606.860352
2010-01-08  10618.190430

In [225]:
djia.shape

(1714, 1)

### Checking Ticket and Saving for - 30 Tickers

In [226]:
for ticket in  ['MSFT', 'PFE', 'MCD', 'UTX', 'DD', 'MMM', 'MRK', 'KO', 'WMT', 'HD', 'AAPL', 'CSCO', \
                'IBM', 'VZ', 'JNJ', 'GE', 'XOM', 'CVX', 'DIS', 'UNH', 'JPM', 'NKE', 'BA', 'CAT', 'TRV',\
                'GS', 'PG', 'V', 'AXP', 'INTC']:
    tickdata[ticket]=web.get_data_yahoo(ticket)

## Q 1. List of Dow companies' stock symbols

In [227]:
dowlist = df({tic:data['Adj Close']
                for tic, data in tickdata.iteritems()})
dowlist.columns

Index([u'AAPL', u'AXP', u'BA', u'CAT', u'CSCO', u'CVX', u'DD', u'DIS', u'DJIA',
       u'GE', u'GS', u'HD', u'IBM', u'INTC', u'JNJ', u'JPM', u'KO', u'MCD',
       u'MMM', u'MRK', u'MSFT', u'NKE', u'PFE', u'PG', u'TRV', u'UNH', u'UTX',
       u'V', u'VZ', u'WMT', u'XOM'],
      dtype='object')

In [228]:
dowlist["Sum"] = 0

In [229]:
dowlist = pd.Series(dowlist.sum(axis=1))

In [230]:
type(dowlist)

pandas.core.series.Series

## Q 2. Correlation value from your index to the actual Dow value

In [231]:
dowlist.corr(djia.DJIA)

0.99993702554784514

## Q 3. Why is this value not exactly 1?  What would be needed to make this value 1?

#### The summation of the 30 Indexes across may have small errors of least count when compared to DJIA. This is exhibited by the correlation factor 0.9999. 

#### In order to make this value 1, we need to have get the exact values in "dowlist" as compared to "djia". 